WordPress/gutenberg
# Elements: Button - Fix element selectors

## What?
In https://github.com/WordPress/gutenberg/pull/41659 we are adding a UI for button elements, which exposes some of the problems in the current approach. When the button element is edited in the Global Styles UI, the correct settings weren't displayed in the Site Editor.

We need to move style for button elements from the CSS for each block into the elements section of theme.json so that the same styles are shared across all elements.

At first I considered putting these rules inside the `__experimentalStyle` key in block.json under the "elements" key, but the problem with this approach is that it makes the rules specific to each block. Then when a user sets the button element styles in Global Styles they would not take effect, because the rules inside each block would take preceidence.

Instead we need the button element rules to live inside the core theme.json file, so that they will be used by all blocks.

## Why?
When a user sets the color for a button element the user's selected color should be applied in the Site Editor. This means that we need to move the rules for button elements to the core theme.json file so that they can be shared by all blocks.

## How?
 This PR attempts to address these issues:
- The styles for the button and file blocks are "beating" the element rules. This moves the button element definitions to the core theme.json file so that they can be overridden by user styles.

- When we call `get_block_data` the core version of the Theme JSON class is instantiated, instead of the Gutenberg version. By using the Gutenberg version we ensure that the element styles are output.

- The button element selector is actually two different selectors. We need to split this by `,` so that elements within buttons can use the correct selectors.

## Important note
This changes the default style of the search block button:

Before:
<img width="506" alt="Screenshot 2022-06-22 at 11 39 28" src="https://user-images.githubusercontent.com/275961/175010007-4cc43194-3e0f-4e55-8a44-a03d9683d230.png">

After:
<img width="507" alt="Screenshot 2022-06-22 at 11 38 56" src="https://user-images.githubusercontent.com/275961/175010059-137d573e-c9b6-49b3-88ab-11a7a3d2741b.png">

Most themes probably won't be affected by this change, as they will already be providing button styles. The only alternative to this that I can see is to provide specific and different button element styles to the search block, which feels unhelpful.

## Testing Instructions
- Use empty theme
- Add a button block, a file block and a search block to your post
- Confirm that the buttons all appear the same (black background, white text) on both the frontend and the editor
- Modify the button colors using theme.json and confirm your changes in both front and back end.
- Modify the button colors using Global Styles and confirm your changes in both front and back end


## Screenshots or screencast 
One nice thing about this approach is that now the default styles are displayed correctly inside Global Styles:

<img width="281" alt="Screenshot 2022-06-20 at 17 56 01" src="https://user-images.githubusercontent.com/275961/174648889-f8fc778d-d766-4301-ac7f-1171623f3fbc.png">


@WordPress/block-themers
